app/services/usps_in_person_proofing/transliterable_validator_helper.rb

```diff
@@ -0,0 +1,88 @@
+module UspsInPersonProofing
+  class TransliterableValidatorHelper
+    SUPPORTED_FIELDS = [
+      :first_name,
+      :last_name,
+      :address1,
+      :address2,
+      :city,
+    ].freeze
+
+    def initialize(transliterator = Transliterator.new)
+      @transliterator = transliterator
+    end
+
+    # Validate that transliteratable fields from the state ID form
+    # can be transliterated to contain acceptable characters.
+    #
+    # @param [Hash] fields Name/value pairs of fields to validate
+    # @option fields [String,#to_s] :first_name
+    # @option fields [String,#to_s] :last_name
+    # @option fields [String,#to_s] :address1
+    # @option fields [String,#to_s] :address2
+    # @option fields [String,#to_s] :city
+    # @return [Hash,nil] Field names mapped to error messages, or nil if there were no issues
+    def validate(fields)
+      validation_result = {}
+      fields.slice(*SUPPORTED_FIELDS).each do |key, value|
+        # Get transliterated value
+        result = @transliterator.transliterate(value)
+        transliterated = result.transliterated
+
+        # Remove question marks corresponding with unsupported chars
+        # for transliteration
+        result.unsupported_chars.each do |chr|
+          transliterated = transliterated.sub(Transliterator::REPLACEMENT, '')
+        end
```

Comment:
I think we could do this in one pass:
```suggestion
        transliterated = transliterated.tr(Transliterator::REPLACEMENT, result.unsupported_chars.join)
```

Reply:
This change fails a test where I expect the replacement character to itself be part of the output.

Input:
```ruby
{
  first_name: "TИb#{replace_char}leЉs",
}
```

```
-:first_name => "Our system cannot read the following characters: ?, Љ, И. Please try again using the characters on your ID.",
+:first_name => "Our system cannot read the following characters: Љ, И. Please try again using the characters on your ID.",
```

Reply:
oh whoops, try this, this was replacing them with `'?'` instead of removing them

```suggestion
        transliterated = transliterated.tr(result.unsupported_chars.join, '')
```

Reply:
That fails several tests where `?` should not appear in the result.

Input:
```ruby
{
  first_name: 'first ДЉ',
  last_name: 'last ЉѲ',
  address1: 'a1 ѮИ',
  address2: 'a2 ЉѺ',
  city: 'city ЧИ',
}
```

```
-:address1 => "Our system cannot read the following characters: И, Ѯ. Please try again using substitutes for those characters.",
-:address2 => "Our system cannot read the following characters: Љ, Ѻ. Please try again using substitutes for those characters.",
-:city => "Our system cannot read the following characters: И, Ч. Please try again using substitutes for those characters.",
-:first_name => "Our system cannot read the following characters: Љ, Д. Please try again using the characters on your ID.",
-:last_name => "Our system cannot read the following characters: Љ, Ѳ. Please try again using the characters on your ID.",
+:address1 => "Our system cannot read the following characters: ?, И, Ѯ. Please try again using substitutes for those characters.",
+:address2 => "Our system cannot read the following characters: ?, Љ, Ѻ. Please try again using substitutes for those characters.",
+:city => "Our system cannot read the following characters: ?, И, Ч. Please try again using substitutes for those characters.",
+:first_name => "Our system cannot read the following characters: ?, Љ, Д. Please try again using the characters on your ID.",
+:last_name => "Our system cannot read the following characters: ?, Љ, Ѳ. Please try again using the characters on your ID.",
```

Reply:
This isn't quite logically equivalent, but it achieves the same overall outcome for the function. I pushed up this to replace the current code.

```ruby
unless transliterated.count(Transliterator::REPLACEMENT) > result.unsupported_chars.length
  transliterated = transliterated.gsub(Transliterator::REPLACEMENT, '')
end
```

Reply:
sounds good!